workflows/templates/workflow-template.yaml

```diff
@@ -0,0 +1,122 @@
+      - name: CEPH_SECRET_KEY
+      - name: CEPH_KEY_ID
+      - name: PYTHONPATH
+
+  templates:
+    - name: srcopsmetrics-workflow
+      steps:
+      - - name: collect-knowledge
+          template: analyse
+
+      - - name: process-information
+          template: process
+
+      - - name: run-dash-viz
+          template: visualize
+
+    - name: analyse
+      container:
+        image: "${IMAGE_STREAM_REGISTRY}/${IMAGE_STREAM_NAMESPACE}/srcopsmetrics:${IMAGE_STREAM_TAG}"
+        command: ["srcopsmetrics/cli.py"]
+        args: ["-c",  "{{inputs.parameters.REPOSITORY}}"]
+        env:
+          - name: GITHUB_ACCESS_TOKEN
+            value: "{{inputs.parameters.GITHUB_ACCESS_TOKEN}}"
+          - name: S3_ENDPOINT_URL
+            value: "{{inputs.parameters.S3_ENDPOINT_URL}}"
+          - name: CEPH_BUCKET
+            value: "{{inputs.parameters.CEPH_BUCKET}}"
+          - name: CEPH_BUCKET_PREFIX
+            value: "{{inputs.parameters}.CEPH_BUCKET_PREFIX}"
+          - name: CEPH_SECRET_KEY
+            value: "{{inputs.parameters.CEPH_SECRET_KEY}}"
+          - name: CEPH_KEY_ID
+            value: "{{inputs.parameters.CEPH_KEY_ID}}"
+          - name: PYTHONPATH
+            value: "."
+
+    - name: process
+      container:
+        image: "${IMAGE_STREAM_REGISTRY}/${IMAGE_STREAM_NAMESPACE}/srcopsmetrics:${IMAGE_STREAM_TAG}"
+        command: ["srcopsmetrics/cli.py"]
+        args: ["-p", "{{inputs.parameters.REPOSITORY}}"]
+        env:
+          - name: GITHUB_ACCESS_TOKEN
+            value: "{{inputs.parameters.GITHUB_ACCESS_TOKEN}}"
+          - name: S3_ENDPOINT_URL
+            value: "{{inputs.parameters.S3_ENDPOINT_URL}}"
+          - name: CEPH_BUCKET
+            value: "{{inputs.parameters.CEPH_BUCKET}}"
+          - name: CEPH_BUCKET_PREFIX
+            value: "{{inputs.parameters}.CEPH_BUCKET_PREFIX}"
+          - name: CEPH_SECRET_KEY
+            value: "{{inputs.parameters.CEPH_SECRET_KEY}}"
+          - name: CEPH_KEY_ID
+            value: "{{inputs.parameters.CEPH_KEY_ID}}"
+          - name: PYTHONPATH
+            value: "."
+
+    - name: visualize
+      container:
+        image: "${IMAGE_STREAM_REGISTRY}/${IMAGE_STREAM_NAMESPACE}/srcopsmetrics:${IMAGE_STREAM_TAG}"
+        command: ["srcopsmetrics/cli.py"]
+        args: ["-v", "{{inputs.parameters.REPOSITORY}}"]
+        env:
+          - name: GITHUB_ACCESS_TOKEN
+            value: "{{inputs.parameters.GITHUB_ACCESS_TOKEN}}"
+          - name: S3_ENDPOINT_URL
+            value: "{{inputs.parameters.S3_ENDPOINT_URL}}"
+          - name: CEPH_BUCKET
+            value: "{{inputs.parameters.CEPH_BUCKET}}"
+          - name: CEPH_BUCKET_PREFIX
+            value: "{{inputs.parameters}.CEPH_BUCKET_PREFIX}"
+          - name: CEPH_SECRET_KEY
+            value: "{{inputs.parameters.CEPH_SECRET_KEY}}"
+          - name: CEPH_KEY_ID
+            value: "{{inputs.parameters.CEPH_KEY_ID}}"
+          - name: PYTHONPATH
+            value: "."
```

Comment:
I don't think we can use this task, the workflow will finish, we won't visualize from the workflow, then we should be able to visualize all this in Superset actually taking the knowledge from Ceph.

Reply:
yes, makes sense